src/librvnpal/inc/rvn.h

```diff
@@ -1,37 +1,109 @@
 #ifndef RVN_H
 #define RVN_H
 
-#include <stdint.h>
-
 #ifdef __APPLE__
-#define EXPORT __attribute__((visibility("default")))	
+#define EXPORT __attribute__((visibility("default")))
+#define PRIVATE __attribute__((visibility("hidden")))
 #elif _WIN32
-#define EXPORT _declspec(dllexport) 
+#define EXPORT _declspec(dllexport)
+#define PRIVATE
 #else
-#define EXPORT 
+#define EXPORT __attribute__((visibility("default")))
+#define PRIVATE __attribute__((visibility("hidden")))
 #endif
 
-typedef int bool;
+#include <stdint.h>
+
+typedef int32_t bool;
 #define true 1
 #define false 0
 
+#define max(x, y) ((x) >= (y)) ? (x) : (y)
+#define min(x, y) ((x) <= (y)) ? (x) : (y)
+
+#define ALLOCATION_GRANULARITY (64*1024)
+
+EXPORT struct SYSTEM_INFORMATION
+{
+    int32_t page_size;
+    int32_t prefetch_status;
+
+    /* can_prefetch => prefetch_status == true */
+};
+
+EXPORT struct RVN_RANGE_LIST
+{
+    void *virtual_address;
+    int32_t number_of_bytes;
+};
+
 EXPORT uint64_t
 rvn_get_current_thread_id(void);
 
 EXPORT int32_t
-rvn_write_header(const char *path, void *header, int32_t size, uint32_t *detailed_error_code);
+rvn_write_header(const char *path, void *header, int32_t size, int32_t *detailed_error_code);
+
+EXPORT int32_t
+rvn_get_error_string(int32_t error, char *buf, int32_t buf_size, int32_t *special_errno_flags);
 
+EXPORT int32_t
+rvn_create_and_mmap64_file(const char *path, int64_t initial_file_size, int32_t flags, void **handle, void **base_addr, int64_t *actual_file_size, int32_t *detailed_error_code);
 
 EXPORT int32_t
-sync_directory_for (const char *file_path, uint32_t * detailed_error_code);
+rvn_prefetch_virtual_memory(void *virtualAddress, int64_t length, int32_t *detailed_error_code);
 
+EXPORT int32_t
+rvn_get_system_information(struct SYSTEM_INFORMATION *sys_info, int32_t *detailed_error_code);
+
+EXPORT int32_t
+rvn_memory_sync(void *address, int64_t size, int32_t *detailed_error_code);
 
-int32_t /* internal use */
-flush_file (int32_t fd);
+EXPORT int32_t
+rvn_dispose_handle(const char *filepath, void *handle, int32_t delete_on_close, int32_t *detailed_error_code);
```

Comment:
```suggestion
rvn_dispose_handle(const char *file_path, void *handle, int32_t delete_on_close, int32_t *detailed_error_code);
```

And we should probably distinguish between journal / pager handles

Reply:
We have `rvn_close_journal` and `rvn_dispose_handle` which is confusing